tests/baselines/reference/transpile/Report an error when compiler-options module-kind is out-of-range.js

```diff
@@ -1 +1,3 @@
+"use strict";
+Object.defineProperty(exports, "__esModule", { value: true });

```

Comment:
This seems like an interesting case; if the module kind is out of range, how are we ending up with CJS?